### PR TITLE
(SVACE) fix major issues (part 1)

### PR DESCRIPTION
**Pull request by grzegorz-czajkowski**
Contains the following commits:
 * BaseKeyboardSupport.js (SVACE): add return to filterNeighbors
 * BaseKeyboardSupport.js (SVACE): add conversion to Boolean
 * BaseKeyboardSupport (SVACE): remove unnecessary argument
 * Graph.js (SVACE): add return value for _build func

### CODE REVIEW

File: src/js/core/widget/BaseKeyboardSupport.js [hunk 1]
```diff
@@ -662,11 +662,11 @@
 						// corner cases, when element positions overlap
 						if (focusableElement.distance === 0) {
 							if (direction === EVENT_POSITION.down) {
-								return currentLink.compareDocumentPosition(typeof focusableElement === element ? focusableElement : focusableElement.element) &
-										Node.DOCUMENT_POSITION_CONTAINED_BY;
+								return !!(currentLink.compareDocumentPosition(typeof focusableElement === element ? focusableElement : focusableElement.element) &
```
Comment (lslachciak) [inline src/js/core/widget/BaseKeyboardSupport.js:665]: Do we always want o have boolean type?
According o func description it returns object
* @return {Object}

**Review by lslachciak — COMMENTED** [review]
one doubt

File: src/js/core/widget/BaseKeyboardSupport.js (hunk 1 above)
Comment (grzegorz-czajkowski) [reply src/js/core/widget/BaseKeyboardSupport.js:665]: This documentation refers to outer function *getNeighborhoodLinks* whereas we are changing callback function for Array.filter where true/false are expected..

Comment (lslachciak) [reply src/js/core/widget/BaseKeyboardSupport.js:665]: right, sorry I overlooked this

**Review by lslachciak — APPROVED** [review]

**Pull request merged**